dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -508,20 +507,29 @@ ModelMulti::evalZ(double t) {
 bool
 ModelMulti::propagateZModif() {
   vector<int> indicesDiff;
-  silentZChange_ = false;
+  silentZChange_.reset();
+  bool nonSilentZChange = false;
   for (int i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
     if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
       indicesDiff.push_back(nonSilentZIndexes_[i]);
+      nonSilentZChange = true;
+    }
+  }
+  for (int i = 0, iEnd = notUsedInContinuousEqSilentZIndexes_.size(); i < iEnd; ++i) {
+    if (doubleNotEquals(zLocal_[notUsedInContinuousEqSilentZIndexes_[i]], zSave_[notUsedInContinuousEqSilentZIndexes_[i]])) {
+      indicesDiff.push_back(notUsedInContinuousEqSilentZIndexes_[i]);
+      if (!silentZChange_.getFlags(NotUsedInContinuousEquations) && !nonSilentZChange)
+        silentZChange_.setFlags(NotUsedInContinuousEquations);
     }
   }
   if (!indicesDiff.empty()) {
     connectorContainer_->propagateZDiff(indicesDiff, zLocal_);
     std::copy(zLocal_, zLocal_ + sizeZ(), zSave_.begin());
     return true;
   } else {
-    for (int i = 0, iEnd = silentZIndexes_.size(); i < iEnd; ++i) {
-      if (doubleNotEquals(zLocal_[silentZIndexes_[i]], zSave_[silentZIndexes_[i]])) {
-        silentZChange_ = true;
+    for (int i = 0, iEnd = notUsedInDiscreteEqSilentZIndexes_.size(); i < iEnd; ++i) {
+      if (doubleNotEquals(zLocal_[notUsedInDiscreteEqSilentZIndexes_[i]], zSave_[notUsedInDiscreteEqSilentZIndexes_[i]])) {
+        silentZChange_.setFlags(NotUsedInDiscreteEquations);
         std::copy(zLocal_, zLocal_ + sizeZ(), zSave_.begin());
```

Comment:
I think I already asked the question but twice is better than none: no easy way to only copy the necessary z instead of the whole vector?

Reply:
it will never be as efficient as using std::copy